### DISCUSSION TITLE
Problems with Engine checks on CI

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#7281] [Work In Progress - CI]
Last several PRs, which did not touch any engine part, were blocked by such error: https://github.com/enso-org/enso/actions/runs/5529969664/jobs/10088750788?pr=7233 

Re-running failed jobs with "clean build" label does not help.

Can someone from engine team take a look? @JaroslavTulach @hubertp ? Or from Standard Library @jdunkerley ?

@mwu-tow can you force-merge https://github.com/enso-org/enso/pull/7233? It has passed all required checks except engine's one, and it does not change anything there.

**Comment by jdunkerley** [comment]
We have had a few cases where it seems like runs are interfering with each other. I haven't seen this specific issue and so far, whenever I have encountered them, a re-run had cleared them out.

**Comment by Akirathan** [comment]
`org.enso.languageserver.websocket.json.TextOperationsTest` seems to fail transiently pretty often - https://github.com/enso-org/enso/actions/runs/5541380729/jobs/10114703713?pr=7257#step:10:6857. cc @4e6 is it possible we have some race conditions there?

**Reply by JaroslavTulach** [reply]
Here is a link to the [failing test case](https://github.com/enso-org/enso/actions/runs/5541380729/jobs/10114703713?pr=7257#step:10:6579).

**Reply by 4e6** [reply]
Probably, I need to look into it. We may want to disable the test for now

**Reply by 4e6** [reply]
Investigating in [#7292](https://github.com/enso-org/enso/pull/7292)

**Comment by Akirathan** [comment]
Another failure that I have seen more than once in the last few weeks/days in `RuntimeSDerverTest` live vs design: https://github.com/enso-org/enso/pull/7276#issuecomment-1634527080

**Comment by farmaazon** [comment]
Another PR with same problem. Re-running didn't help: https://github.com/enso-org/enso/actions/runs/5621801982/job/15284640461?pr=7337

I still don't understand the issue here. Engine tests are failing, but they are not required to merge my PR. However, this strange "reporter" _is_ required and often complains about missing tests results.

**Reply by Akirathan** [reply]
This failure is expected. 

The `Run ./run backend ci-check` does two kinds of tests - Engine tests and Standard Library tests. If Engine tests fail, the Standard Library tests are not invoked at all, and therefore, `Standard Library Test Reporter` step fails, as there are, obviously, no test reports. The action run you linked fails in Engine tests - https://github.com/enso-org/enso/actions/runs/5621801982/job/15284640461?pr=7337#step:10:6304. 

The action run fails because `Run ./run backend ci-check` step failed, not because `Standard Library Test Reporter` step failed.
